http.ts

```diff
@@ -121,8 +125,80 @@ export class ServerRequest {
   method: string;
   proto: string;
   headers: Headers;
+  conn: Conn;
+  r: BufReader;
   w: BufWriter;
 
+  public async *readChunk() {
+    if (this.headers.has("content-length")) {
+      const len = +this.headers.get("content-length");
+      if (Number.isNaN(len)) {
+        return new Uint8Array(0);
+      }
+      let buf = new Uint8Array(1024);
+      let rr = await this.r.read(buf);
+      let nread = rr.nread;
+      while (!rr.eof && nread < len) {
+        yield buf.subarray(0, rr.nread);
+        buf = new Uint8Array(1024);
+        rr = await this.r.read(buf);
+        nread += rr.nread;
+      }
+      yield buf.subarray(0, rr.nread);
+    } else if (this.headers.get("transfer-encoding") === "chunked") {
+      // Based on https://tools.ietf.org/html/rfc2616#section-19.4.6
+      const tp = new TextProtoReader(this.r);
+      let [line, _] = await tp.readLine();
+      // TODO: handle chunk extension
+      let [chunkSizeString, optExt] = line.split(";");
+      let chunkSize = parseInt(chunkSizeString, 16);
+      if (Number.isNaN(chunkSize) || chunkSize < 0) {
+        throw new Error("Invalid chunk size");
```

Comment:
Should errors while reading chunked body result in `ERR_INVALID_CHUNKED_ENCODING`?

Reply:
Hmm we haven't actually formalize error types yet...